inference-engine/src/gna_plugin/transformations/insert_transpose_after_convolution_or_pooling.cpp

```diff
@@ -0,0 +1,108 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/insert_transpose_after_convolution_or_pooling.hpp"
+
+#include <numeric>
+
+#include <ngraph/opsets/opset1.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+#include "gna_plugin_log.hpp"
+
+using namespace GNAPluginNS;
+
+NGRAPH_RTTI_DEFINITION(InsertTransposeAfterConvOrPool, "InsertTransposeAfterConvOrPool", 0);
+
+InsertTransposeAfterConvOrPool::InsertTransposeAfterConvOrPool() {
+    auto conv_or_pool = ngraph::pattern::wrap_type<ngraph::opset1::Convolution, ngraph::opset1::MaxPool>();
+    auto reshape_const = ngraph::pattern::wrap_type<ngraph::opset1::Constant>();
+    auto reshape = ngraph::pattern::wrap_type<ngraph::opset1::Reshape>({conv_or_pool, reshape_const});
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher &m) {
+        auto& pattern_map = m.get_pattern_value_map();
+        auto conv_or_pool_node = pattern_map.at(conv_or_pool).get_node_shared_ptr();
+        auto next_node = pattern_map.at(reshape).get_node_shared_ptr();
+        bool found_reshape_to_1d = false;
+        std::shared_ptr<ngraph::Node> reshape_node = next_node;
+        std::shared_ptr<ngraph::Node> transpose_node = nullptr;
+        while ((reshape_node != nullptr || transpose_node != nullptr) && next_node->get_output_size() == 1) {
```

Comment:
Looks like this transformation doesn't fit well in the matcher pass concept. In my mind, we should not have any additional search cycles inside. Maybe it should be a function pass? 

cc @GlebKazantaev

Reply:
I agree, replaced by a function pass